clis/teliod/src/main.rs

```diff
@@ -0,0 +1,240 @@
+}
+
+struct CommandListener {
+    socket: DaemonSocket,
+}
+
+impl CommandListener {
+    async fn get_command(&self) -> Result<EventLoopCommand, String> {
+        let mut connection = match self.socket.accept().await {
+            Ok(conn) => conn,
+            Err(err) => {
+                return Err(format!(
+                    "Failed to accept connection: {:?} - stopping daemon",
+                    err
+                ));
+            }
+        };
+
+        let Ok(command_str) = connection.read_command().await else {
+            return Err("Unknown command".to_string());
+        };
+
+        if let Ok(cmd) = serde_json::from_str::<Cmd>(&command_str) {
+            match cmd {
+                Cmd::Daemon { .. } => {
+                    if connection.respond("ERR".to_owned()).await.is_err() {
+                        Err("Socket failed on sending response".to_string())
+                    } else {
+                        Err("Start command should never be passed to the daemon".to_string())
+                    }
+                }
+                Cmd::HelloWorld { name } => {
+                    if connection.respond("OK".to_owned()).await.is_err() {
+                        Err("Socket failed on sending response".to_string())
+                    } else {
+                        Ok(EventLoopCommand::HelloWorld(name))
+                    }
+                }
+            }
+        } else if connection.respond("ERR".to_owned()).await.is_err() {
+            Err("Socket failed on sending response".to_string())
+        } else {
+            Err("Received unknown command".to_string())
+        }
+    }
+}
+
+// From async context Telio needs to be run in separate task
+fn telio_task() {
+    let _telio = Device::new(
+        Features::default(),
+        // TODO: replace this with some real event handling
+        move |event| info!("Incoming event: {:?}", event),
+        None,
+    );
+}
+
+async fn daemon_event_loop(config: TeliodDaemonConfig) -> Result<()> {
+    let (non_blocking_writer, _tracing_worker_guard) =
+        tracing_appender::non_blocking(fs::File::create(config.log_file_path)?);
+    tracing_subscriber::fmt()
+        .with_max_level(config.log_level)
+        .with_writer(non_blocking_writer)
+        .with_ansi(false)
+        .with_line_number(true)
+        .with_level(true)
+        .init();
+
+    let socket = DaemonSocket::new(&TeliodDaemon::get_ipc_socket_path()?)?;
+    let cmd_listener = CommandListener { socket };
+
+    let telio_task_handle = tokio::task::spawn_blocking(telio_task);
+
+    let mut signals = Signals::new([SIGTERM])?;
+
+    info!("Entering event loop");
+
+    let err_str = loop {
+        select! {
+            maybe_cmd = cmd_listener.get_command() => {
+                match maybe_cmd {
+                    Ok(EventLoopCommand::HelloWorld(name)) => {
+                        info!("Hello {}", name);
+                    }
+                    Err(err_str) => {
```

Comment:
I do not think we should exit the daemon if client sends invalid command. IMHO, lets just log the error, and respond to a client with some kind of error response.